### PR TITLE
Generalize readinessManager to handle liveness

### PR DESCRIPTION
**Pull request by timstclair**
In preparation for liveness, where the ResultsManager will be used outside the prober package. See https://github.com/kubernetes/kubernetes/pull/15275 for planned usage.

### CODE REVIEW

**Comment by timstclair** [comment]
Failure looks unrelated.

File: pkg/kubelet/prober/results/results_manager.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+	defer m.RUnlock()
+	result, found := m.cache[id]
+	return result, found
+}
+
+func (m *manager) Set(id kubecontainer.ContainerID, result Result) {
+	m.Lock()
+	defer m.Unlock()
+	prev, exists := m.cache[id]
+	if !exists || prev != result {
```
Comment (yujuhong) [inline pkg/kubelet/prober/results/results_manager.go]: Why checking here? This is equivalent to `m.cache[id] = result`

Comment (timstclair) [inline pkg/kubelet/prober/results/results_manager.go]: It's an artifact of the way I extracted this out of the liveness workers PR.  The missing context is [here](https://github.com/timstclair/kubernetes/commit/adbe8f23ca51d19ac82f5949733f4ea9be556cc1#diff-cfbedfd685dbb1af1a45ddd3175d4f0eR94). Let me know whether you still want me to change it.

Comment (yujuhong) [inline pkg/kubelet/prober/results/results_manager.go]: I thought that might be the case. I am ok with it for now.

**Comment by wojtek-t** [comment]
@k8s-bot test this please

**Pull request merged**